From 82010950651b3225d6bc16c2c8b1929ad958e40e Mon Sep 17 00:00:00 2001
From: Ken Zhu <quic_guigenz@quicinc.com>
Date: Tue, 20 Jun 2023 13:16:31 -0700
Subject: [PATCH 316/500] linux: fix icmp of ipip6 tunnel not working issue

It should use address of the header instead of the address of point
of the address of header

Change-Id: I2c183c80c2cb0e0b41c933d2548919f4c3a34baa
Signed-off-by: Zhu Ken <guigenz@codeaurora.org>
Signed-off-by: John Sanli <quic_jsanli@quicinc.com>
---
 net/ipv6/ip6_tunnel.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/net/ipv6/ip6_tunnel.c b/net/ipv6/ip6_tunnel.c
index 9340392d5e3d..b92c3c41a7a9 100644
--- a/net/ipv6/ip6_tunnel.c
+++ b/net/ipv6/ip6_tunnel.c
@@ -849,7 +849,7 @@ static void ip4ip6_fmr_calc(struct in6_addr *dest,
 	use_dest_addr = (xmit && dsth == iph) || (!xmit && dsth != iph);
 
 	/* get dst port */
-	if (((u8*)&dsth[1]) <= end && (
+	if (((u8 *)&dsth[1]) <= end && (
 		dsth->protocol == IPPROTO_UDP ||
 		dsth->protocol == IPPROTO_TCP ||
 		dsth->protocol == IPPROTO_SCTP ||
@@ -864,9 +864,9 @@ static void ip4ip6_fmr_calc(struct in6_addr *dest,
 		struct icmphdr *ih = (struct icmphdr*)(((u8*)dsth) + dsth->ihl * 4);
 
 		/* use icmp identifier as port */
-		if (((u8*)&ih) <= end && (
-		    (use_dest_addr && (
-		    ih->type == ICMP_ECHOREPLY ||
+		if (((u8 *)ih) <= end && (
+			(use_dest_addr && (
+			ih->type == ICMP_ECHOREPLY ||
 			ih->type == ICMP_TIMESTAMPREPLY ||
 			ih->type == ICMP_INFO_REPLY ||
 			ih->type == ICMP_ADDRESSREPLY)) ||
-- 
2.34.1

